monai/transforms/transforms.py

```diff
@@ -206,10 +207,112 @@ def __call__(self, img):
+        self.rotate_params = rotate_params
+        self.shear_params = shear_params
+        self.translate_params = translate_params
+        self.scale_params = scale_params
+
+    def __call__(self, spatial_size, grid=None):
+        """
+        Args:
+            spatial_size (list or tuple of int): output grid size.
+            grid (ndarray): grid to be transformed. Shape must be (3, H, W) for 2D or (4, H, W, D) for 3D.
+        """
+        if grid is None:
+            grid = create_grid(spatial_size)
+
+        spatial_dims = len(spatial_size)
+        affine = np.eye(spatial_dims + 1)
+        if self.rotate_params:
+            affine = affine @ create_rotate(spatial_dims, self.rotate_params)
+        if self.shear_params:
+            affine = affine @ create_shear(spatial_dims, self.shear_params)
+        if self.translate_params:
+            affine = affine @ create_translate(spatial_dims, self.translate_params)
+        if self.scale_params:
+            affine = affine @ create_scale(spatial_dims, self.scale_params)
+        return (affine @ grid.reshape(grid.shape[0], -1)).reshape([-1] + list(spatial_size))
+
+
+class Resample:
+
+    def __init__(self, padding_mode='reflection', as_tensor_output=False, device=None):
+        """
+        computes output image using values from `img`, locations from `grid`.
+
+        Args:
+            padding_mode ('zeros'|'border'|'reflection'): mode of handling out of range indices. Defaults to 'zeros'.
+            as_tensor_output(bool): whether to return a torch tensor. Defaults to False.
+            device (string):
+        """
+        self.padding_mode = padding_mode
+        self.as_tensor_output = as_tensor_output
+        self.device = device
+
+    def __call__(self, img, grid, mode='bilinear'):
+        """
+        Args:
+            img (ndarray or tensor): shape must be (num_channels, H, W[, D]).
+            grid (ndarray or tensor): shape must be (3, H, W) for 2D or (4, H, W, D) for 3D.
+            mode ('nearest'|'bilinear'): interpolation order. Defaults to 'bilinear'.
+        """
+        if not isinstance(img, torch.Tensor):
+            img = torch.tensor(img)
+        if not isinstance(grid, torch.Tensor):
+            grid = torch.tensor(grid)
+        if self.device:
+            img.to(self.device)
+            grid.to(self.device)
+
+        for i, dim in enumerate(img.shape[1:]):
+            grid[i] = 2. * grid[i] / (dim - 1.)
+        grid = grid[:-1] / grid[-1:]
+        grid = grid[range(img.ndim - 2, -1, -1)]
+        grid = grid.permute(list(range(grid.ndim))[1:] + [0])
+        new_img = torch.nn.functional.grid_sample(img[None], grid[None], mode=mode, padding_mode=self.padding_mode)[0]
+        if not self.as_tensor_output:
+            return new_img.cpu().numpy()
+        return new_img
+
+
+class Affine:
+    """
+    transform ``img`` given the affine parameters.
+    """
+
+    def __init__(self,
```

Comment:
Are the transform parameters randomly chosen at training time? I don't see how that is meant to happen quite yet.

Reply:
it's not in this PR yet, I'm working on it..